plugins/modules/aws_glue_job.py

```diff
@@ -254,6 +283,19 @@ def _compare_glue_job_params(user_params, current_params):
     return False
 
 
+def _get_glue_job_arn(module, job_name):
+    (region, account_id) = (module.region, _get_aws_account_id(module))
+    return "arn:aws:glue:" + region + ":" + account_id + ":" + "job/" + job_name
```

Comment:
The `lambda` module tries to construct the ARN too. You may want to take a look at https://github.com/ansible-collections/community.aws/pull/5/files, and see how that's being done for that module. Perhaps that code should be made a common utility.

Reply:
We have `ansible_collections.amazon.aws.plugins.module_utils.iam.get_aws_account_id()` which is a slightly more robust version of `_get_aws_account_id`.

There's also https://github.com/ansible-collections/amazon.aws/pull/136 (currently pending unit test creation) which would also provide the partition (`arn:aws` vs `arn:aws-gov` vs `arn:aws-cn`).